### ISSUE TITLE
Numeric overflow in `OverLimits` check

### ISSUE THREAD

**Issue by tomcur** [slskd/slskd#1174]
Since updating from 0.19 to 0.21.1 I'm getting these messages in the slskd log with debug messages enabled:

```
11:26:10	INF	Rejected enqueue request for user "****": Queued limits exceeded
11:26:10	DBG	Projected bytes 22010062 exceeds limit -294967296
```

This negative limit seems suspect. I have configured the limits as:

```
global:
  upload:
    slots: 3
    speed_limit: 4096
  limits:
    queued:
      files: 300
      megabytes: 4000
    daily:
      files: 5000
      megabytes: 30000
      failures: 200
    weekly:
      files: 10000
      megabytes: 60000
      failures: 600
```

The `OverLimits` function calculates `byteLimit * 1000 * 1000`. Which for a queued limit of 4,000 megabytes is 4,000,000,000. This doesn't fit in a signed 32-bit integer, overflowing to -294967296. So it appears this is the source of the issue.

https://github.com/slskd/slskd/blob/bce1f91af62a8b3269f1e477e2a2311bcad2112f/src/slskd/Application.cs#L580

If it matters, this is on a Linux build, but I suspect the same overflow occurs on builds for other platforms.

**Comment by jpdillingham** [comment]
Thanks for reporting this, I should be able to get a fix out this evening.

**Comment by jpdillingham** [comment]
Bit distracted this week, this is my first priority tomorrow morning.

**Comment by jpdillingham** [comment]
Fixed in [0.21.3](https://github.com/slskd/slskd/releases/tag/0.21.3).  Thanks again for reporting the issue!
